artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/RemotingConnectionImpl.java

```diff
@@ -391,6 +391,15 @@ public void bufferReceived(final Object connectionID, final ActiveMQBuffer buffe
       }
    }
 
+   @Override
+   public void endOfBatch(Object connectionID) {
+      super.endOfBatch(connectionID);
+      // TODO we really need a lock here?
```

Comment:
Can you clear this TODO before we merge it?

Reply:
That's my only point on this PR.

Reply:
I don't have my laptop around, but I can try do it tomorrow somehow...
The answer in the to-do is still valid: do we need a lock there?